extensions/azurePublish/src/node/index.ts

```diff
@@ -150,6 +150,95 @@ export default async (composer: IExtensionRegistration): Promise<void> => {
     /* These methods implement the publish actions */
     /*******************************************************************************************************************************/
     /**
+     * Prepare a bot to be built and deployed by copying the runtime and declarative assets into a temporary folder
+     * @param project
+     * @param settings
+     * @param srcTemplate
+     * @param resourcekey
+     */
+    private init = async (project: any, srcTemplate: string, resourcekey: string, runtime: any) => {
```

Comment:
why are you using all sync version of all the IO functions?

Reply:
beacuse io functions may be casue race condition